snuba/schemas.py

```diff
@@ -238,10 +207,48 @@
             ],
             'minItems': 3,
             'maxItems': 3,
-        }
+        },
     }
 }
 
+EVENTS_QUERY_SCHEMA = {
+    **copy.deepcopy(GENERIC_QUERY_SCHEMA),
+    # Need to select down to the project level for customer isolation and performance
+    'required': ['project'],
+}
+
+EVENTS_QUERY_SCHEMA['properties'].update({
+    **get_time_series_query_schema_properties(
+        default_granularity=3600,
+        default_window=timedelta(days=5),
+    ),
+    'project': {
+        'anyOf': [
+            {'type': 'number'},
+            {
+                'type': 'array',
+                'items': {'type': 'number'},
+                'minItems': 1,
+            },
+        ]
+    },
+    # Never add FINAL to queries, enable sampling
+    'turbo': {
+        'type': 'boolean',
+        'default': False,
+    },
+    # Force queries to hit the first shard replica, ensuring the query
+    # sees data that was written before the query. This burdens the
+    # first replica, so should only be used when absolutely necessary.
+    'consistent': {
+        'type': 'boolean',
+        'default': False,
+    },
+    'debug': {
+        'type': 'boolean',
+    }
```

Comment:
Some of these might eventually find their way back into the generic query options, but I think it's best to promote them back upwards rather than trying to optimize in the other direction if possible.